services/uploads-scan/clamav.js

```diff
@@ -0,0 +1,139 @@
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const execSync = require('child_process').execSync;
+const path = require('path');
+const constants = require('./constants');
+const utils = require("./utils");
+
+const S3 = new AWS.S3();
+
+/**
+ * Updates the definitions using freshclam.
+ *
+ * It will download the definitions to the current work dir.
+ */
+function updateAVDefinitonsWithFreshclam() {
+    try {
+        let executionResult = execSync(`${constants.PATH_TO_FRESHCLAM} --config-file=${constants.FRESHCLAM_CONFIG} --datadir=${constants.FRESHCLAM_WORK_DIR}`);
+        
+        utils.generateSystemMessage('Update message');
+        console.log(executionResult.toString());
+
+        if(executionResult.stderr) {
+            utils.generateSystemMessage('stderr');
+            console.log(executionResult.stderr.toString());
+        }
+
+        return true;
+    } catch (err) {
+        console.log(err);
+        return false;
+    }
+
+}
+
+/**
+ * Download the Antivirus definition from S3.
+ * The definitions are stored on the local disk, ensure there's enough space.
+ */
+async function downloadAVDefinitions() {
+
+    const downloadPromises = constants.CLAMAV_DEFINITIONS_FILES.map((filenameToDownload) => {
+       return new Promise((resolve, reject) => {
+           let destinationFile = path.join('/tmp/', filenameToDownload);
+
+           utils.generateSystemMessage(`Downloading ${filenameToDownload} from S3 to ${destinationFile}`);
+
+           let localFileWriteStream = fs.createWriteStream(destinationFile);
+
+           let options = {
+               Bucket: constants.CLAMAV_BUCKET_NAME,
+               Key   : `${constants.PATH_TO_AV_DEFINITIONS}/${filenameToDownload}`
+           };
+
+           let s3ReadStream = S3.getObject(options).createReadStream().on('end', function () {
+               utils.generateSystemMessage(`Finished download ${filenameToDownload}`);
+               resolve();
+           }).on('error', function (err) {
+               utils.generateSystemMessage(`Error downloading definition file ${filenameToDownload}`);
+               console.log(err);
+               reject();
+           });
+
+           s3ReadStream.pipe(localFileWriteStream);
+       });
+    });
+
+    return await Promise.all(downloadPromises);
+}
+
+/**
+ * Uploads the AV definitions to the S3 bucket.
+ */
+async function uploadAVDefinitions() {
+
+    const uploadPromises = constants.CLAMAV_DEFINITIONS_FILES.map((filenameToUpload) => {
+        return new Promise((resolve, reject) => {
+            utils.generateSystemMessage(`Uploading updated definitions for file ${filenameToUpload} ---`);
+
+            let options = {
+                Bucket: constants.CLAMAV_BUCKET_NAME,
+                Key   : `${constants.PATH_TO_AV_DEFINITIONS}/${filenameToUpload}`,
+                Body  : fs.createReadStream(path.join('/tmp/', filenameToUpload)),
+                ACL   : 'public-read'
```

Comment:
Like above, I think this can be private, since your lambda role's grant permission